tailscale/resource_tailnet_key.go

```diff
@@ -167,7 +165,11 @@ func resourceTailnetKeyRead(ctx context.Context, d *schema.ResourceData, m inter
 		return diagnosticsError(err, "Failed to set created_at")
 	}
 
-	if err = d.Set("expires_at", key.Expires.Format(time.RFC3339)); err != nil {
+	if key.Expires.Before(time.Now().UTC()) {
```

Comment:
As you pointed out, this seems a bit risky, since we can't guarantee if the client's local time is in sync, so we might accidentally be deleting non-expired keys from Terraform state.

A slightly better way to handle this might be to compare `expires` field with the `date` HTTP response header, but it's not trivial to implement since the response headers are not currently exposed to the provider.

Or perhaps we could explicitly mark keys as expired in the response? Or not return expired keys at all?